Add movement key presses to the HUD.

oitzujoey opened the pull request:
This adds an option to show arrow keys on the HUD like Defrag does. The use of this is that players can watch each other's key presses when learning new movement.

This adds three client cvars
`cg_hudMovementKeys`: Set to 1 to show key presses on the HUD.
`cg_hudMovementKeysScale`: Sets the size of the arrows.
`cg_hudMovementKeysColor`: Sets the HSV color of the arrows.

This patch uses six new TGA files and a modified HUD shader. These can be found on Discord in the file "arrow-keys-hud.pk3".

oitzujoey committed:
Add movement key presses to the HUD.

oitzujoey commented:
To get the key presses to the client, I used another `pm->ps->stats` element. Is there a better way to do this?

rdntcntrl commented:
Ah I see, this is intended mostly for spectators or people watching demos? We will run out of those stats fields pretty soon at this rate. They are precious ;)

One small thing: It would be nice if the assets don't have to be loaded unless someone enables this. The easiest way of doing that is to set the CVAR_LATCH flag which makes it require a restart to change. Then you can check if it's even enabled before loading the assets. 
Other features also track if the user changed a cvar and then load it, which is more convenient to the player but requires a check every frame.

rdntcntrl commented:
I don't necessarily have a better way of doing it. I am a bit hesitant about the stats fields since this is not a feature necessary for the core gameplay. But any other option would be an ugly workaround. 

You could for instance send a separate, invisible event-type entity to players who have the feature enabled. But it would be ugly. We probably should only start doing that once we really need to.

oitzujoey committed:
Change cg_hudMovementKeys to CVAR_LATCH.

oitzujoey commented:
I have added the attribute CVAR_LATCH to cg_hudMovementKeys.

I will try to make it so that a game restart isn't required since I think this cvar will be enabled for only short periods of time, and it would be a nuisance to restart the game twice within five minutes. To do that, do I just run `trap_R_RegisterShaderNoMip("movementKeyIndicatorDown")` when the cvar is first enabled? I'm guessing that I'm only supposed to do this only once per run of the game?

oitzujoey committed:
Allow players to view movement keys on HUD without restarting.

Remove CVAR_LATCH from cg_hudMovementKeys but still only register
shaders the first time the cvar is set.

oitzujoey commented:
I removed `CVAR_LATCH` and now load the shaders the first time `cg_hudMovementKeys` is set.
